### PR TITLE
Use "iso:" prefix for fiat currency codes

### PR DESCRIPTION
**Pull request by peachbits**

### CODE REVIEW

File: src/config.ts [hunk 1]
```diff
@@ -27,7 +27,7 @@ export const asConfig = asObject({
   httpPort: asOptional(asNumber, 8008),
```
Comment (peachbits) [inline src/config.ts:27]: add iso:USD constant

File: src/config.ts [hunk 2]
```diff
@@ -61,10 +61,10 @@ export const asConfig = asObject({
     'VET'
   ]),
   fiatCurrencyCodes: asOptional(asArray(asString), [
```
Comment (peachbits) [inline src/config.ts:63]: add to config "DEFAULT_FIAT_CODE"

File: src/utils/currencyCodeMaps.ts [hunk 3]
```diff
@@ -25,10 +25,10 @@ export const zeroRateCurrencyCodes = {
   FORK: true
```
Comment (peachbits) [inline src/utils/currencyCodeMaps.ts:25]: Make json

File: src/utils/currencyCodeMaps.ts [hunk 4]
```diff
@@ -25,10 +25,10 @@ export const zeroRateCurrencyCodes = {
...
 }
 export const fallbackConstantRatePairs = {
-  SAI_USD: '1',
-  DAI_USD: '1',
-  TESTBTC_USD: '0.01',
-  BRZ_BRL: '1'
+  'SAI_iso:USD': '1',
```
Comment (peachbits) [inline src/utils/currencyCodeMaps.ts:28]: create function to add isos to constants

File: src/providers/coinMarketCap.ts [hunk 5]
```diff
@@ -66,7 +71,7 @@ const query = async (
     rates[date] = {}
     Object.keys(json.data).forEach(code => {
       if (json.data[code].quotes.length > 0)
-        rates[date][`${code}_iso:USD`] = json.data[
+        rates[date][`${code}_${DEFAULT_FIAT}`] = json.data[
```
Comment (thehobbit85) [inline src/providers/coinMarketCap.ts:74]: Create and use the function everywhere in the code:
```
const defualtFiatPair = (code) => `${code}_${DEFAULT_FIAT}`
```

File: src/providers/currencyConverter.ts [hunk 6]
```diff
@@ -77,15 +82,15 @@ const currencyConverter = async (
     const toCurrency = pair.currency_pair.split('_')[1]
     if (
       isFiatCode(fromCurrency) &&
-      fromCurrency !== 'iso:USD' &&
+      fromCurrency !== DEFAULT_FIAT &&
       datesAndCodesWanted[pair.date].indexOf(`${subIso(fromCurrency)}_USD`) ===
```
Comment (thehobbit85) [inline src/providers/currencyConverter.ts:87]: This can be something like `${subIso(fromCurrency)}_${subIso(DEFAULT_FIAT)}` no?

**Review by thehobbit85 — CHANGES_REQUESTED** [review]
See comments.

File: src/providers/coinMarketCap.ts [hunk 7]
```diff
@@ -37,37 +39,35 @@ const OPTIONS = {
...
-        ...out,
-        [`${code}_${DEFAULT_FIAT}`]: results.data[code].quotes[0].quote[
-          subIso(DEFAULT_FIAT)
-        ].price.toString()
-      }
-    }, {})
+const coinMarketCapHistoricalQuote = (
+  data: ReturnType<typeof ascoinMarketCapHistoricalQuotes>,
+  code: string
+): string => data[code].quotes[0].quote[subIso(DEFAULT_FIAT)].price.toString()
```
Comment (peachbits) [inline src/providers/coinMarketCap.ts:65]: dont use specific key in array. use an array method or ?

File: src/utils/utils.ts [hunk 8]
```diff
@@ -91,11 +85,59 @@ export const logger = (...args): void => {
...
+
+export const toCode = (pair: string): string => currencyCodeArray(pair)[1]
+
+export const invertPair = (pair: string): string =>
+  `${toCode(pair)}_${fromCode(pair)}`
+
+export const createReducedRateMapArray = (
+  createCurrencyPair: IsoOpObject,
+  createCurrencyQuote: (code: any) => string
+) => (data: any[]): RateMap =>
```
Comment (thehobbit85) [inline src/utils/utils.ts:124]: Instead of typing any, use generics:
```<T>(data: T[]): RateMap =>....```

File: src/utils/utils.ts [hunk 9]
```diff
@@ -91,11 +85,59 @@ export const logger = (...args): void => {
...
+    return {
+      ...out,
+      [createCurrencyPair(code)]: createCurrencyQuote(code)
+    }
+  }, {})
+
+export const createReducedRateMap = (
+  createCurrencyPair: IsoOp,
+  createCurrencyQuote: (rates, code: string) => string
+) => (data): RateMap =>
```
Comment (thehobbit85) [inline src/utils/utils.ts:135]: use generics, I think this can work, not 100% sure ;)

```<T extends {}>(data: T): RateMap =>....```

**Review by thehobbit85 — CHANGES_REQUESTED** [review]
Some type fixing

**Pull request merged**